src/client/pages/Section/DataTable/Table/RowData/Cell/hooks/useOnChange/useOnChangeNodeValue.ts

```diff
@@ -0,0 +1,167 @@
+    value,
+  })
+
+  const enablerDependants = AssessmentMetaCaches.getEnablersDependants({
+    assessment,
+    cycle,
+    tableName,
+    variableName,
+  })
+
+  return enablerDependants.filter((dep) => {
+    const depRow = table.rows?.find((r) => r.props.variableName === dep.variableName)
+    const depCol = depRow?.cols?.find((c) => c.props.colName === dep.colName)
+    if (!depRow || !depCol) return false
+    const enableIf = depCol.props.enableIf?.[cycle.uuid]
+    if (!enableIf) return false
+
+    const rowCache: RowCache = {
+      ...depRow,
+      tableName,
+      sectionName,
+    }
+
+    const enabled = ExpressionEvaluator.evalFormula<boolean>({
+      assessment,
+      assessments: { [assessment.props.name]: assessment },
+      countryIso,
+      cycle,
+      data: futureData,
+      colName: depCol.props.colName,
+      row: rowCache,
+      formula: enableIf,
+    })
+
+    const depValue = RecordAssessmentDatas.getNodeValue({
+      assessmentName: assessment.props.name,
+      cycleName: cycle.name,
+      data,
+      countryIso,
+      tableName,
+      colName: depCol.props.colName,
+      variableName: depRow.props.variableName,
+    })
+
+    return !enabled && depValue && depValue.raw
+  })
+}
+
+export const useOnChangeNodeValue = (props: Props) => {
+  const { col, row, sectionName, table } = props
+
+  const dispatch = useAppDispatch()
+  const countryIso = useCountryIso()
+  const cycle = useCycle()
+  const assessment = useAssessment()
+  const assessmentSection = useSection(sectionName)
+
+  const { t } = useTranslation()
+  const persistSanitizedValue = usePersistSanitizedValue(props)
+
+  return useCallback<OnChangeNodeValue>(
+    (value: NodeValue): void => {
+      const tableName = table.props.name
+      const { variableName } = row.props
+      const { colName } = col.props
+
+      const affected = getAffectedNodes({ ...props, assessment, countryIso, cycle, value })
+
+      // If node change affects other row x cols:
+      if (affected.length > 0) {
+        const fields = affected.map((a) => t(`measures.${a.variableName}`)).join('\n')
+
+        const confirmed = window.confirm(t('common.areYouSureFollowingFieldsWillBeDisabled', { fields }))
+
+        // Do nothing if the user cancels the confirmation
+        if (!confirmed) return
+
+        const updates = [
+          {
+            colName,
+            value,
+            variableName,
+          },
+          ...affected.map((a) => ({
+            colName: a.colName,
+            value: { raw: null } as NodeValue,
+            variableName: a.variableName,
+          })),
+        ]
+
+        dispatch(
+          DataActions.updateNodeValues({
+            assessmentName: assessment.props.name,
+            cycleName: cycle.name,
+            sectionName: assessmentSection?.props.name,
+            countryIso,
+            tableName,
+            values: updates,
+          })
+        )
+        return
+      }
```

Comment:
Main change 1
(Rest of the changes is just splitting the useOnChange to separate files)